server/middleware/user.middleware.ts

```diff
@@ -0,0 +1,135 @@
+import { UserService } from "../services/user.service";
+
+const userService = new UserService();
+
+export class UserMiddleware {
+  async varifySignupRequestBody(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void> {
+    try {
+      // Extract user data from request body
+      const { email, name, password }: IResisterRequestbody = req.body;
+
+      // validate name, email and password
+      if (!name)
+        return responseProvider.sendResponse({
+          message: "Bad Request | Invalid name",
+          response: res,
+          statusCode: 400,
+        });
+
+      if (!email)
+        return responseProvider.sendResponse({
+          message: "Bad Request | Invalid email",
+          response: res,
+          statusCode: 400,
+        });
+      if (!password)
+        return responseProvider.sendResponse({
+          message: "Bad Request | Invalid password",
+          response: res,
+          statusCode: 400,
+        });
+
+      next();
+    } catch (err) {
+      return responseProvider.InternalServerError({
+        response: res,
+        error: err as Error,
+      });
+    }
+  }
+  async varifyLoginRequestBody(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void> {
+    try {
+      // Extract user data from request body
+      const { email, password }: ILoginRequestbody = req.body;
+
+      // validate email and password
+      if (!email)
+        return responseProvider.sendResponse({
+          message: "Bad Request | Invalid email",
+          response: res,
+          statusCode: 400,
+        });
+      if (!password)
+        return responseProvider.sendResponse({
+          message: "Bad Request | Invalid password",
+          response: res,
+          statusCode: 400,
+        });
+
+      next();
+    } catch (err) {
+      return responseProvider.InternalServerError({
+        response: res,
+        error: err as Error,
+      });
+    }
+  }
+  async varifyAuthToken(
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<void> {
+    try {
+      // Extract user data from headers
+      const { authorization } = req.headers;
+
+      // validate authorization
+      if (!authorization) {
+        return responseProvider.sendResponse({
+          message: "Bad Request | Authorization header is missing",
+          response: res,
+          statusCode: 400,
+        });
+      }
+
+      // Extract token from authorization
+      const token = authorization.split(" ")[1];
+
+      // validate token
+      if (!token) {
+        return responseProvider.sendResponse({
+          message: "Bad Request | Invalid Token",
+          response: res,
+          statusCode: 400,
+        });
+      }
+      const isValidToken = await tokenHandler.verifyToken(token);
+
+      // validate active user with token
+
+      const validateActiveUser = await userService.findUserData({
+        access_token: token,
+      });
```

Comment:
no need to find active user from database we can get userId from token only

Reply:
And why are we storing access-token in the database? there is no need of that

Reply:
done both